server/handlers/component_handler.go

```diff
@@ -1178,6 +1179,59 @@ func prettifyCompDefSchema(entities []entity.Entity) []v1beta1.ComponentDefiniti
 	return comps
 }
 
+// swagger:route POST /api/meshmodel/export ExportModel idExportModel
+// Handle GET request for exporting a model.
+//
+// Export model with the given id in the output format specified
+//
+// ```?id={id}```
+// ```?output_format={output_format}``` Can be `json`, `yaml`, or `oci`. Default is `oci`
+//
+// responses:
+//	200: []byte
+func (h *Handler) ExportModel(rw http.ResponseWriter, r *http.Request) {
+	modelId := r.URL.Query().Get("id")
+
+	// 1. Get the model data
+	modelFilter := &regv1beta1.ModelFilter{Id: modelId}
+	e, err := h.registryManager.GetEntityById(modelFilter)
+	if err != nil {
+		h.log.Error(ErrGetMeshModels(err)) //TODO: Add appropriate meshkit error
+		http.Error(rw, ErrGetMeshModels(err).Error(), http.StatusInternalServerError)
+	}
+	model := e.(*v1beta1.Model)
+
+	// 2. Convert it to oci
+	temp := os.TempDir()
+	modelDir := filepath.Join(temp, model.Name)
+	os.Mkdir(modelDir, 0700)
+	defer os.RemoveAll(modelDir)
+	err = model.WriteModelDefinition(filepath.Join(modelDir, "model.yaml"), "yaml")
+	// build oci image for the model
+	img, err := oci.BuildImage(modelDir)
+	if err != nil {
+		h.log.Error(ErrGetMeshModels(err)) //TODO: Add appropriate meshkit error
+		http.Error(rw, ErrGetMeshModels(err).Error(), http.StatusInternalServerError)
+	}
+	tarfileName := filepath.Join(modelDir, "model.tar")
+	err = oci.SaveOCIArtifact(img, tarfileName, model.Name)
```

Comment:
Is archive compression happening here? If not, where is the `.gz` bit happening?

Reply:
Ideally, we're being consistent with behavior used while exporting Designs (same function, same options). Using MeshKit to achieve this? https://github.com/meshery/meshkit/blob/fd9565f3bbbfb6731c25726714029913f82b3743/models/oci/utils.go#L27